mantidproject/mantid
# Enable parsing TOML wav ranges

**Description of work.**

- Adds Dataclass backport (this commit is also in #29805)
- Enables parsing TOML wavelength ranges
- Consolidates the logic into a helper by moving the existing layer out of the GUI layer
- More tests + system tests for wavelength reduction in TOML files

**To test:**
- Check tests run, the system tests include a TOML file with the new syntax
- Download the ISIS training data
- Open the SANS GUI
- For user file either use maskfile.toml or maskfile.txt
- Go to settings -> Q / Wavelength...etc.
- Under wavelength change from `Lin` to `RangeLin` (or Log to RangeLog)
- For the binning enter `1-3, 4-10`
- Open a batch file and point to the csv in LOQDemo
- Select first row and hit process selected. The output workspace should have `1-3` and `4-10` in their names representing the wavelength limits

Part of #28987 

Release notes TODO



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
